functions/go/starlark/go.mod

```diff
@@ -4,5 +4,8 @@ go 1.16
 
 require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
-	sigs.k8s.io/kustomize/kyaml v0.10.21
+	k8s.io/api v0.21.0
+	k8s.io/apimachinery v0.21.0
+	sigs.k8s.io/kustomize/kyaml v0.11.1-0.20210630195303-1eb3c1a07570
```

Comment:
Using kyaml from HEAD to pick up https://github.com/kubernetes-sigs/kustomize/pull/4023.
We should change to a released version when it is released.

Reply:
Which functionality in current PR is effected if the referred change in kyaml is not picked up ?

Reply:
`metadata.name` will be lost after decoding.
And `metadata.name` will be used as `Name` in the `starlark.Filter`: https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/457/files#diff-45dd66dfd0fba255d8026ad222e5a8b78be5bb608fc4d6d7da177aea735a7b0eR120 And `Name` is a required field there.

Any concerns?